_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -599,8 +599,7 @@ mfxStatus VideoDECODEVP9_HW::GetOutputSurface(mfxFrameSurface1 **surface_out, mf
     mfxFrameSurface1 *pNativeSurface = m_FrameAllocator->GetSurface(index, surface_work, &m_vInitPar);
     if (pNativeSurface)
     {
-        mfxFrameSurface1 *pOpaqueSurface = m_core->GetOpaqSurface(pNativeSurface->Data.MemId);
-        *surface_out = pOpaqueSurface ? pOpaqueSurface : pNativeSurface;
+        *surface_out = m_is_opaque_memory ? m_core->GetOpaqSurface(pNativeSurface->Data.MemId) : pNativeSurface;
```

Comment:
This looks like core returns not NULL pointer to opaque surface while opaque memory is not used (`m_is_opaque_memory` == false). Is this improper usage of opaque memory in component or misbehavior of core?